### ISSUE TITLE
`get` with destination stream does not `end` when remote file does not exist

### ISSUE THREAD

**Issue by mhassan1** [theophilusx/ssh2-sftp-client#427]
`sftp.get('i-do-not-exist', writable)` does not call `writable.end()`; as a result, the writable remains open forever, which could lead to a file descriptor leak.

To replicate:
1. Run a script like this that does not exit:
```js
await sftp.connect()
const writable = fs.createWriteStream('my-local-file')
try {
  await sftp.get('i-do-not-exist', writable) // -> throws because remote file does not exist
} catch (ignore) {}
await setTimeout(1e6)
```
2. Then, run `lsof -p <pid> | grep my-local-file` to see the leftover file descriptor.

**Comment by mhesham93** [comment]
I also have the exact same problem, using the latest version.

**Comment by theophilusx** [comment]
and the same response!

I need a basic script which reproduces the issue as well as details
aboutg node version and platform.

To be very clear, I am using this functionality daily with 3 different
systems with no issue. Unless I can reproduce the issue, I cannot
diagnose it or assist in finding a resolution.

**Comment by theophilusx** [comment]
Apologies, I got this issue mixed up with a different issue.

I have a tentative fix for this one, but am not happy with it and need
to do a lot more testing and I don't have time right now.

This won't occur for at least 2 weeks.

**Comment by theophilusx** [comment]
Fixed and version 9.0.3 pushed to npmjs.com
